### PR TITLE
KAA-1203: Add unit test

### PR DESCRIPTION
**Pull request by RostakaGmfun**
@rasendubi @forGGe @Polinaa

### CODE REVIEW

File: client/client-multi/client-c/test/test_kaa_channel_manager.c [hunk 1]
```diff
@@ -458,7 +464,24 @@ void test_get_bootstrap_client_sync_serialize(void **state)
...
+
+    error_code = kaa_channel_manager_create(&channel_manager, &kaa_context);
+    ASSERT_EQUAL(error_code, KAA_ERR_NONE);
+    kaa_channel_manager_set_auth_failure_handler(channel_manager, test_auth_failure_handler, &reason);
+
+    kaa_channel_manager_process_auth_failure(channel_manager, KAA_AUTH_STATUS_BAD_CREDENTIALS);
+    ASSERT_EQUAL(reason, KAA_AUTH_STATUS_BAD_CREDENTIALS);
+
+    kaa_channel_manager_process_auth_failure(channel_manager, KAA_AUTH_STATUS_VERIFICATION_FAILED);
+    ASSERT_EQUAL(reason, KAA_AUTH_STATUS_VERIFICATION_FAILED);
```
Comment (rasendubi) [inline client/client-multi/client-c/test/test_kaa_channel_manager.c]: Don't use the macro. Have you read our [testing guide](https://github.com/kaaproject/kaa/blob/develop/client/client-multi/client-c/docs/testing.org)?

**Comment by RostakaGmfun** [comment]
Done, @rasendubi, re-review please

**Comment by forGGe** [comment]
+

File: client/client-multi/client-c/test/test_kaa_channel_manager.c [hunk 2]
```diff
@@ -458,7 +464,24 @@ void test_get_bootstrap_client_sync_serialize(void **state)
...
+void test_process_auth_failure(void **state)
+{
+    (void)state;
+
+    kaa_error_t error_code;
+    kaa_channel_manager_t *channel_manager = NULL;
+    kaa_auth_failure_reason reason;
+
+    error_code = kaa_channel_manager_create(&channel_manager, &kaa_context);
+    assert_return_code(error_code, KAA_ERR_NONE);
```
Comment (forGGe) [inline client/client-multi/client-c/test/test_kaa_channel_manager.c]: You misunderstood `assert_return_code()` purpose. Please, check [documentation](https://api.cmocka.org/group__cmocka__asserts.html#ga9370e705f22fa98b18664bacb9c525e8). Generally, `assert_int_equal()` should be used.

**Comment by forGGe** [comment]
Small nitpick

**Comment by forGGe** [comment]
+

**Comment by Oldnickus** [comment]
ok to test

**Comment by Oldnickus** [comment]
test this please

**Comment by Oldnickus** [comment]
retest this please

**Comment by Oldnickus** [comment]
test this please

**Comment by Oldnickus** [comment]
test this please

**Comment by Oldnickus** [comment]
@kaa-leeroy test this please

**Pull request merged**